plugins/adrv9002.c

```diff
@@ -603,30 +604,80 @@ static void check_nco_support(struct plugin_private *priv)
 	}
 }
 
-static void load_profile(GtkFileChooser *chooser, gpointer data)
+static char *read_file(const char *file, ssize_t *f_size)
 {
-	struct plugin_private *priv = data;
-	char *file_name = gtk_file_chooser_get_filename(chooser);
 	FILE *f;
 	char *buf;
-	int ret;
 	ssize_t size;
 
-	f = fopen(file_name, "r");
+	f = fopen(file, "r");
 	if (!f)
-		goto err;
+		return NULL;
 
 	fseek(f, 0, SEEK_END);
 	size = ftell(f);
 	rewind(f);
+
 	buf = malloc(size);
 	if (!buf) {
 		fclose(f);
-		goto err;
+		return NULL;
+	}
+
+	*f_size = fread(buf, sizeof(char), size, f);
+	if (*f_size < size) {
+		free(buf);
+		buf = NULL;
 	}
 
-	size = fread(buf, sizeof(char), size, f);
 	fclose(f);
+	return buf;
+}
+
+static void load_stream(GtkFileChooser *chooser, gpointer data)
+{
+	struct plugin_private *priv = data;
+	char *file_name = gtk_file_chooser_get_filename(chooser);
+	char *buf;
+	ssize_t size;
+	int ret;
+
+	buf = read_file(file_name, &size);
+	if (!buf)
+		goto err;
+
+	ret = iio_device_attr_write_raw(priv->adrv9002, "stream_config", buf, size);
+	free(buf);
+	if (ret < 0)
+		goto err;
+
+	gtk_file_chooser_set_filename(chooser, file_name);
+	strncpy(priv->last_stream, file_name, sizeof(priv->last_stream) - 1);
+	g_free(file_name);
+	return;
+err:
+	g_free(file_name);
+	dialog_box_message(GTK_WIDGET(chooser), "Stream Loading Failed",
+			   "Failed to load stream using the selected file!");
+
+	if (priv->last_stream[0])
+		gtk_file_chooser_set_filename(chooser, priv->last_stream);
+	else
+		gtk_file_chooser_set_filename(chooser, "(None)");
+}
+
+static void load_profile(GtkFileChooser *chooser, gpointer data)
+{
+	struct plugin_private *priv = data;
+	char *file_name = gtk_file_chooser_get_filename(chooser);
```

Comment:
is this missing a `g_free(file_name);`?

Reply:
It should be freed... Is it not? The only change I've make to this function was to use the new helper. But it might a bug lying there from the beginning...

Reply:
oh; it is;
apologies;
the patch is a bit wonky too look at